test/Atmos/EDMF/dycore_kernels.jl

```diff
@@ -0,0 +1,289 @@
+#### Dynamical core kernels
+
+# Reference state
+abstract type ReferenceState end
+struct HydrostaticState{P, FT} <: ReferenceState
+    virtual_temperature_profile::P
+    relative_humidity::FT
+end
+function HydrostaticState(
+    virtual_temperature_profile::TemperatureProfile{FT},
+) where {FT}
+    return HydrostaticState{typeof(virtual_temperature_profile), FT}(
+        virtual_temperature_profile,
+        FT(0),
+    )
+end
+
+# Specify auxiliary variables for `SingleStack`
+
+function vars_state_auxiliary(m::SingleStack, FT)
+    @vars(z::FT,
+          buoyancy::FT,
+          ρ0::FT,
+          p0::FT,
+          edmf::vars_state_auxiliary(m.edmf, FT),
+          # ref_state::vars_state_auxiliary(m.ref_state, FT) # quickly added at end
+          );
+end
+
+function vars_state_conservative(m::SingleStack, FT)
+    @vars(ρ::FT,
+          ρu::SVector{3, FT},
+          ρe_int::FT,
+          ρq_tot::FT,
+          edmf::vars_state_conservative(m.edmf, FT));
+end
+
+function vars_state_gradient(m::SingleStack, FT)
+    @vars(p0::FT,
+          e_int::FT,
+          q_tot::FT,
+          u::FT,
+          edmf::vars_state_gradient(m.edmf, FT));
+end
+
+function vars_state_gradient_flux(m::SingleStack, FT)
+    @vars(∇e_int::SVector{3, FT},
+          ∇q_tot::SVector{3, FT},
+          ∇u::SMatrix{3, 3, FT, 9},
+          ∇p0::SVector{3, FT},
+          edmf::vars_state_gradient_flux(m.edmf, FT));
+end
+# ## Define the compute kernels
+
+# Specify the initial values in `aux::Vars`, which are available in
+# `init_state_conservative!`. Note that
+# - this method is only called at `t=0`
+# - `aux.z` and `aux.T` are available here because we've specified `z` and `T`
+# in `vars_state_auxiliary`
+function init_state_auxiliary!(m::SingleStack{FT,N}, aux::Vars, geom::LocalGeometry) where {FT,N}
+    aux.z = geom.coord[3]
+    T_profile = DecayingTemperatureProfile{FT}(m.param_set)
+    ref_state = HydrostaticState(T_profile)
+    T_virt, p = ref_state.virtual_temperature_profile(m.param_set, aux.z)
+    _R_d = FT(R_d(m.param_set))
+
+    aux.buoyancy = 0
+    aux.ρ0 = p / (_R_d * T_virt)
+    aux.p0 = p
+
+    init_state_auxiliary!(m, m.edmf, aux, geom)
+end;
+
+function update_auxiliary_state!(
+    dg::DGModel,
+    m::SingleStack,
+    Q::MPIStateArray,
+    t::Real,
+    elems::UnitRange,
+)
+
+    return true
+end
```

Comment:
```suggestion
```

This method is redefined below

Reply:
Do you mean in 107  ? I was sure this duplicate is needed to assign this function to the EDMF subcomponents (updrafts, env) ..
which one should I keep ?